src/Zed/Common/Factory/StatelessFactoryRule.php

```diff
@@ -1,52 +0,0 @@
-<?php
-
-namespace ArchitectureSniffer\Zed\Common\Factory;
-
-use PHPMD\AbstractNode;
-use PHPMD\AbstractRule;
-use PHPMD\Node\ClassNode;
-use PHPMD\Rule\ClassAware;
-
-/**
- * Factories shouldn't contain any property. Properties would make the factory stateful.
- */
-class StatelessFactoryRule extends AbstractRule implements ClassAware
```

Comment:
The rule is invalid and utterly wrong, the caching of (immutable) factories and builders is exactly doing the opposite of what the text here claims: It actually prevents statefulness.
For factories the properties must stay possible.